tests/rptest/tests/e2e_topic_recovery_test.py

```diff
@@ -0,0 +1,171 @@
+        """Stop all redpanda nodes"""
+        for node in self.redpanda.nodes:
+            self.logger.info(f"Node {node.account.hostname} will be stopped")
+            if not node is self._verifier_node:
+                self.redpanda.stop_node(node)
+        time.sleep(10)
+
+    def _start_redpanda_nodes(self):
+        """Start all redpanda nodes"""
+        for node in self.redpanda.nodes:
+            self.logger.info(f"Starting node {node.account.hostname}")
+            if not node is self._verifier_node:
+                self.redpanda.start_node(node)
+        time.sleep(10)
+
+    def _wipe_data(self):
+        """Remove all data from redpanda cluster"""
+        for node in self.redpanda.nodes:
+            self.logger.info(
+                f"All data will be removed from node {node.account.hostname}")
+            self.redpanda.remove_local_data(node)
+
+    def _restore_topic(self, topic_spec, overrides={}):
+        """Restore individual topic"""
+        self.logger.info(f"Restore topic called. Topic-manifest: {topic_spec}")
+        conf = {
+            'redpanda.remote.recovery': 'true',
+            #'redpanda.remote.write': 'true',
+        }
+        conf.update(overrides)
+        self.logger.info(f"Confg: {conf}")
+        topic = topic_spec.name
+        npart = topic_spec.partition_count
+        nrepl = topic_spec.replication_factor
+        rpk = RpkTool(self.redpanda)
+        rpk.create_topic(topic, npart, nrepl, conf)
+        time.sleep(10)
+        rpk.describe_topic(topic)
+        rpk.describe_topic_configs(topic)
+
+    @cluster(num_nodes=4)
+    @matrix(message_size=[10000],
+            num_messages=[100000],
+            recovery_overrides=[{}, {
+                'retention.bytes': 1024
+            }])
+    def test_restore(self, message_size, num_messages, recovery_overrides):
+        """Write some data. Remove local data then restore
+        the cluster."""
+
+        if self.debug_mode:
+            self.logger.info(
+                "Skipping test in debug mode (requires release build)")
+            return
+
+        self.init_producer(message_size, num_messages)
+        self._producer.start(clean=False)
+
+        self._producer.wait()
+        assert self._producer.produce_status.acked >= num_messages
+
+        def s3_has_all_data():
+            objects = list(self.redpanda.get_objects_from_si())
+            total_size = 0
+            for o in objects:
+                if o.Key.endswith("/manifest.json") and self.topic in o.Key:
+                    data = self.redpanda.s3_client.get_object_data(
+                        self._bucket, o.Key)
+                    manifest = json.loads(data)
+                    last_upl_offset = manifest['last_offset']
+                    self.logger.info(
+                        f"Found manifest at {o.Key}, last_offset is {last_upl_offset}"
+                    )
+                    return last_upl_offset >= num_messages
```

Comment:
That's a bit incorrect (kafka vs redpanda offsets again), but I guess good enough. Alternatively we can look at the max timestamp.